hugr-py/src/hugr/hugr/render.py

```diff
@@ -266,6 +284,7 @@ def _viz_node(self, node: Node, hugr: Hugr, graph: Digraph) -> None:
                     self._viz_node(child, hugr, sub)
                 html_label = self._format_html_label(**label_config)
                 sub.node(f"{node.idx}", shape="plain", label=f"<{html_label}>")
+                self.nodes.add(node)
```

Comment:
Maybe hoist this out of the if statement since you're doing the same in both branches?

Reply:
Done.